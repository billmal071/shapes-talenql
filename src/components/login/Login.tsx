import React, { useContext, useEffect, useState } from 'react';
import { useHistory } from 'react-router-dom';
import Swal from "sweetalert2";
import LoginContext from '../../context/login/loginContext';
import styles from "./login.module.scss";

function Login() {
  const history = useHistory();
  const [username, setUsername] = useState<string>("Test1");
  const [password, setPassword] = useState<string>("TalentQL");
  const { login, isLoggedIn } = useContext(LoginContext)

  async function handleSubmit(e: React.FormEvent<HTMLFormElement>) {
    e.preventDefault();

    if (username.trim() === "") {
      console.log("no username")
      await Swal.fire({
        icon: "error",
        text: "Enter a username"
      })
    }

    if (password.trim() === "") {
      console.log("no password")
      await Swal.fire({
        icon: "error",
        text: "Enter a password"
      })
    }

    if (username.match("Test1") && password.match("TalentQL")) {
      console.log(password);
      login({ username, password })
    } else {
      console.log("wrong credentials")
      await Swal.fire({
        icon: "error",
        text: "Wrong Credentials"
      })
    }
  }

  useEffect(() => {
    if (isLoggedIn) {
      history.push("/")
    }
  }, [isLoggedIn])

  return (
    <div className={`${styles.h__90vh} ${styles.login}`}>
      <h4 className="uppercase center-text f__35 mt__30">login</h4>
      <form data-testid="loginForm" className={styles.login__body} onSubmit={(e) => handleSubmit(e)}>
        <div>
          <label htmlFor="username" className={`${styles.login__text} capitalize`}>username</label>
          <input data-testid="username" type="text" name="username" id="username" className={`${styles.login__input}`} value={username} onChange={(e) => setUsername(e.target.value)} placeholder="type username here..." required />
        </div>
        <div>
          <label htmlFor="password" className={`${styles.login__text} capitalize`}>password</label>
          <input data-testid="password" type="password" name="password" id="password" className={`${styles.login__input}`} value={password} onChange={(e) => setPassword(e.target.value)} placeholder="*********" required />
        </div>
        <div className="center-text">
          <button data-testid="submit" type="submit" className={`${styles.login__btn} uppercase`}>login</button>
        </div>
      </form>
    </div>
  )
}

export default Login
