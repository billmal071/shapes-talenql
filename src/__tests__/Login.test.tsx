import React from 'react';
import { render, fireEvent } from "@testing-library/react";
import renderer from 'react-test-renderer';

import Login from '../components/login/Login';

/**
 * @jest-environment jsdom
 */

describe("<Login />", () => {
  test("should display a login form", async () => {
    const { findByTestId } = render(<Login />);
    const loginForm = await findByTestId("loginForm");

    expect(loginForm).toHaveFormValues({
      username: "Test1",
      password: "TalentQL",
    })
  })

  test("should allow entering a username", async () => {
    const { getByTestId } = render(<Login />);

    fireEvent.change(getByTestId("username"), { target: { value: "Test1" } })

    expect(getByTestId("username")).toHaveValue("Test1")
  })

  test("should allow entering a password", async () => {
    const { getByTestId, findByRole } = render(<Login />);

    fireEvent.change(getByTestId("password"), { target: { value: "TalentQL" } })

    expect(getByTestId("password")).toHaveValue("TalentQL")
  })

  test("should submit the form with username and password", async () => {
    const handleSubmit = jest.fn();
    const { getByTestId } = render(<Login />);
    const { getInstance, toJSON } = renderer.create(<Login />);

    let tree = toJSON();
    expect(tree).toMatchSnapshot();

    fireEvent.change(getByTestId("username"), { target: { value: "Test1" } })
    fireEvent.change(getByTestId("password"), { target: { value: "TalentQL" } })
    fireEvent.click(getByTestId("submit"))

    expect(getByTestId("submit")).toBeEnabled()
  })
})