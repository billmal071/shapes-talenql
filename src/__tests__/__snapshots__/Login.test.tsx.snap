// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Login /> should submit the form with username and password 1`] = `
<div
  className="undefined undefined"
>
  <h4
    className="uppercase center-text f__35 mt__30"
  >
    login
  </h4>
  <form
    data-testid="loginForm"
    onSubmit={[Function]}
  >
    <div>
      <label
        className="undefined capitalize"
        htmlFor="username"
      >
        username
      </label>
      <input
        className="undefined"
        data-testid="username"
        id="username"
        name="username"
        onChange={[Function]}
        placeholder="type username here..."
        required={true}
        type="text"
        value="Test1"
      />
    </div>
    <div>
      <label
        className="undefined capitalize"
        htmlFor="password"
      >
        password
      </label>
      <input
        className="undefined"
        data-testid="password"
        id="password"
        name="password"
        onChange={[Function]}
        placeholder="*********"
        required={true}
        type="password"
        value="TalentQL"
      />
    </div>
    <div
      className="center-text"
    >
      <button
        className="undefined uppercase"
        data-testid="submit"
        type="submit"
      >
        login
      </button>
    </div>
  </form>
</div>
`;
